DOC/projections: Simplify links in the projection table to use directly the titles of the examples

yvonnefroehlich opened the pull request:
**Description of proposed changes**

See https://github.com/GenericMappingTools/pygmt/issues/3405#issue-2476920122 for context.

- [x] Correct link in projection table to projection code **A** (the first row)
- [x] Simply link in projection table and use titles of examples
- [x] Make projection names  consistent - identical case: lower-case
- [x] Make projection names consistent - keep / add term "projection"
- [x] Make projection names consistent - identical term: following the [headings of the GMT documentation](https://docs.generic-mapping-tools.org/dev/reference/map-projections.html)

Related upstream GMT PR: https://github.com/GenericMappingTools/gmt/pull/8567

**Preview**
- projection table: https://pygmt-dev--3407.org.readthedocs.build/en/3407/techref/projections.html
- projection examples: https://pygmt-dev--3407.org.readthedocs.build/en/3407/projections/index.html

Fixes #3405


**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

seisman commented on doc/techref/projections.md:
```diff
@@ -16,34 +16,34 @@ The table below shows the projection codes for the 31 GMT map projections:
+| **Oa**{{ lon0 }}/{{ lat0 }}/*azimuth*/*width*[**+v**]       | {doc}`/projections/cyl/cyl_oblique_mercator_1` |
+| **Ob**{{ lon0 }}/{{ lat0 }}/{{ lon1 }}/{{ lat1 }}/*width*[**+v**] | {doc}`/projections/cyl/cyl_oblique_mercator_2` |
+| **Oc**{{ lon0 }}/{{ lat0 }}/{{ lonp }}/{{ latp }}/*width*[**+v**] | {doc}`/projections/cyl/cyl_oblique_mercator_3` |
+| **P***width*[**+a**][**+f**[**e**\|**p**\|*radius*]][**+r***offset*][**+t***origin*][**+z**[**p**\|*radius*]] | {doc}`Polar </projections/nongeo/polar` [azimuthal] ({math}`\theta, r`) (or cylindrical) |
```
```suggestion
| **P***width*[**+a**][**+f**[**e**\|**p**\|*radius*]][**+r***offset*][**+t***origin*][**+z**[**p**\|*radius*]] | {doc}`Polar </projections/nongeo/polar>` [azimuthal] ({math}`\theta, r`) (or cylindrical) |
```

yvonnefroehlich replied:
The link is fixed in commit https://github.com/GenericMappingTools/pygmt/pull/3407/commits/81e161981c272eb47ad3fb1e59d6cce2df53f378.

seisman commented on doc/techref/projections.md:
```diff
@@ -16,34 +16,34 @@ The table below shows the projection codes for the 31 GMT map projections:
+| **U***zone*/*width*                                         | {doc}`/projections/cyl/cyl_universal_transverse_mercator` |
+| **V**[{{ lon0 }}/]*width*                                   | {doc}`/projections/misc/misc_van_der_grinten` |
+| **W**[{{ lon0 }}/]*width*                                   | {doc}`/projections/misc/misc_mollweide` |
+| **X***width*[**l**\|**p***exp*\|**T**\|**t**][/*height*[**l**\|**p***exp*\|**T**\|**t**]][**d**] | {doc}`/projections/nongeo/cartesian_linear`, {doc}`/projections/nongeo/cartesian_logarithmic`, {doc}`/projections/nongeo/cartesian_power`, and time |
```
Now the "projection name" for "-JX" is "Cartesian linear, Cartesian logarithmic, Cartesian power, and time". It's a little too long. 

I think for `-JX`, we should use the original version for simplicity.

yvonnefroehlich replied:
The titles are shorten in commit https://github.com/GenericMappingTools/pygmt/pull/3407/commits/6d9fb0525418e666fe3c845a0aa30574359dfb70.

seisman commented on examples/projections/misc/misc_mollweide.py:
```diff
@@ -1,6 +1,6 @@
 r"""
-Mollweide
```
Let's wait for the upstream change first.

yvonnefroehlich replied:
So the GMT-related PR is merged. Let's continue here.
Do we want to make the projection names consistent with GMT or consistent between the overview table and the examples? For the later one, do we want to use the name from the GMT projection table or form the headings of the GMTdocumentation page?

seisman replied:
I think we should make the projection names consistent with GMT.

yvonnefroehlich replied:
I have made a few adjustments in commit https://github.com/GenericMappingTools/pygmt/pull/3407/commits/013381eef2a7e08b7f137a5e7f9eb93f891908ac.

seisman commented:
Not sure if we can finish the upstream PR https://github.com/GenericMappingTools/gmt/pull/8567 and this PR in time. If not, at least we can fix the link to the projection code **A** in PR #3370, and then work on the projection table after v0.13.0. @yvonnefroehlich What do you think?

yvonnefroehlich commented:
> Not sure if we can finish the upstream PR [GenericMappingTools/gmt#8567](https://github.com/GenericMappingTools/gmt/pull/8567) and this PR in time. If not, at least we can fix the link to the projection code **A** in PR #3370, and then work on the projection table after v0.13.0. @yvonnefroehlich What do you think?

Yes, unfortunately it looks like I am running a bit out of time with these PRs. For now, I fixed the like to the projection code **A** in the common typo fixes PR #3370.

yvonnefroehlich commented:
> > Not sure if we can finish the upstream PR [GenericMappingTools/gmt#8567](https://github.com/GenericMappingTools/gmt/pull/8567) and this PR in time. If not, at least we can fix the link to the projection code **A** in PR #3370, and then work on the projection table after v0.13.0. @yvonnefroehlich What do you think?
> 
> Yes, unfortunately it looks like I am running a bit out of time with these PRs (just back to work since this Monday). For now, I fixed the like to the projection code **A** in the common typo fixes PR #3370.

Maybe we can split the PR into two parts: (i) simplifying the links (this PR, and completed for v0.13.0) and (ii) making the projection names consistent (new PR, and moved to v0.14.0).

seisman commented:
> Maybe we can split the PR into two parts: (i) simplifying the links (this PR, and completed for v0.13.0) and (ii) making the projection names consistent (new PR, and moved to v0.14.0).

Sounds good to me.

> I have reverted the changes of the projection names (see [fb2e904](https://github.com/GenericMappingTools/pygmt/commit/fb2e904a78619e112824cb97ad7ab41a986f8f60)). For now, the names are only changed regarding using consistently lower-case letters and removing the term "projection" (commit [46b37b7](https://github.com/GenericMappingTools/pygmt/commit/46b37b7e6edb19e9835e52c42f9cd7ea119933f3)).
I'm debating if we should keep the "projection" term instead. For some projections like Mercator or Hammer, the term "projection" makes no difference, but for some projections like "Polyconic", "Polyconic" is an adjective, and "Polyconic projection" may make more sense in the example title.

yvonnefroehlich commented:
> > I have reverted the changes of the projection names (see [fb2e904](https://github.com/GenericMappingTools/pygmt/commit/fb2e904a78619e112824cb97ad7ab41a986f8f60)). For now, the names are only changed regarding using consistently lower-case letters and removing the term "projection" (commit [46b37b7](https://github.com/GenericMappingTools/pygmt/commit/46b37b7e6edb19e9835e52c42f9cd7ea119933f3)).

> I'm debating if we should keep the "projection" term instead. For some projections like Mercator or Hammer, the term "projection" makes no difference, but for some projections like "Polyconic", "Polyconic" is an adjective, and "Polyconic projection" may make more sense in the example title.

Yes, I was actually wondering about this. But as we currently have "Azimuthal equistant", "General stereographic", "Gnomonic", "Orthographic", "Equidistant conic", and "Sinusoidal" without the "projection" and these are also adjectives and sound (at least for me) better with the term "projection", I removed it for "Polyconic" and "Lambert conic conformal" to make it consistent. Maybe we can include this in the projection names discussion and decide on the GMT docs first and do the same for the PyGMT docs?

yvonnefroehlich commented:
> > > I have reverted the changes of the projection names (see [fb2e904](https://github.com/GenericMappingTools/pygmt/commit/fb2e904a78619e112824cb97ad7ab41a986f8f60)). For now, the names are only changed regarding using consistently lower-case letters and removing the term "projection" (commit [46b37b7](https://github.com/GenericMappingTools/pygmt/commit/46b37b7e6edb19e9835e52c42f9cd7ea119933f3)).
> 
> > I'm debating if we should keep the "projection" term instead. For some projections like Mercator or Hammer, the term "projection" makes no difference, but for some projections like "Polyconic", "Polyconic" is an adjective, and "Polyconic projection" may make more sense in the example title.
> 
> Yes, I was actually wondering about this. But as we currently have "Azimuthal equistant", "General stereographic", "Gnomonic", "Orthographic", "Equidistant conic", and "Sinusoidal" without the term "projection" and these are also adjectives and sound (at least for me) better with "projection", I removed it for "Polyconic" and "Lambert conic conformal" to make it consistent. Maybe we can include this in the projection names discussion and decide on the GMT docs first (see also comment [GenericMappingTools/gmt#8567 (comment)](https://github.com/GenericMappingTools/gmt/pull/8567#issuecomment-2303055568)) and do the same for the PyGMT docs?

Hm. I now reverted all projection name changes in this PR.

> > > I have reverted the changes of the projection names (see [fb2e904](https://github.com/GenericMappingTools/pygmt/commit/fb2e904a78619e112824cb97ad7ab41a986f8f60)). For now, the names are only changed regarding using consistently lower-case letters and removing the term "projection" (commit [46b37b7](https://github.com/GenericMappingTools/pygmt/commit/46b37b7e6edb19e9835e52c42f9cd7ea119933f3)).
> 
> > I'm debating if we should keep the "projection" term instead. For some projections like Mercator or Hammer, the term "projection" makes no difference, but for some projections like "Polyconic", "Polyconic" is an adjective, and "Polyconic projection" may make more sense in the example title.
> 
> Yes, I was actually wondering about this. But as we currently have "Azimuthal equistant", "General stereographic", "Gnomonic", "Orthographic", "Equidistant conic", and "Sinusoidal" without the term "projection" and these are also adjectives and sound (at least for me) better with "projection", I removed it for "Polyconic" and "Lambert conic conformal" to make it consistent. Maybe we can include this in the projection names discussion and decide on the GMT docs first (see also comment [GenericMappingTools/gmt#8567 (comment)](https://github.com/GenericMappingTools/gmt/pull/8567#issuecomment-2303055568)) and do the same for the PyGMT docs?

Hm. I am currently unsure what still has to be done in this PR. Should I revert all changes of the projection names, including the change to lower-case? Or should I only add the term "projection" back for for "Polyconic" and "Lambert conic conformal"?

seisman commented:
I prefer to leave this PR to v0.14.0 so that we can have more comments/thoughts from others before we make more changes.

yvonnefroehlich commented:
> I prefer to leave this PR to v0.14.0 so that we can have more comments/thoughts from others before we make more changes.

That's fine with me.

yvonnefroehlich commented:
\format

seisman commented:
Is this PR finished? I haven't checked all changes carefully, but generally they look good.

One off-topic question: Do you think we should combine the three gallery examples for the Oblique Mercator projections (https://www.pygmt.org/dev/projections/cyl/cyl_oblique_mercator_1.html, https://www.pygmt.org/dev/projections/cyl/cyl_oblique_mercator_2.html, https://www.pygmt.org/dev/projections/cyl/cyl_oblique_mercator_3.html) into the same page, since they are just one projection with different ways to specify the projection parameters.

yvonnefroehlich commented on examples/projections/misc/misc_eckertVI.py:
```diff
@@ -1,6 +1,6 @@
 r"""
-Eckert VI
-=========
+Eckert VI projection
```
```suggestion
Eckert VI equal-area projection
```

yvonnefroehlich commented on examples/projections/misc/misc_eckertIV.py:
```diff
@@ -1,6 +1,6 @@
 r"""
-Eckert IV
-=========
+Eckert IV projection
```
```suggestion
Eckert IV equal-area projection
```

yvonnefroehlich commented on examples/projections/azim/azim_general_perspective.py:
```diff
@@ -1,6 +1,6 @@
 r"""
-General Perspective
-===================
+General perspective projection
+==============================
```
```suggestion
Perspective projection
======================
```

yvonnefroehlich commented on examples/projections/azim/azim_general_perspective.py:
```diff
@@ -1,6 +1,6 @@
+General perspective projection
+==============================
 
 The general perspective projection imitates the view of the Earth from a finite
 point in space. In a full view of the earth one third of its surface area can
```
```suggestion
The perspective projection imitates the view of the Earth from a finite
point in space. In a full view of the earth one third of its surface area can
```

yvonnefroehlich commented:
> Is this PR finished? I haven't checked all changes carefully, but generally they look good.

I think now all names should be consistent with the headings of the GMT documentation.

seisman commented on examples/projections/misc/misc_eckertVI.py:
```diff
@@ -1,6 +1,6 @@
 r"""
-Eckert VI
-=========
+Eckert VI equal-area projection
+====================
```
```suggestion
Eckert VI equal-area projection
===============================
```

seisman commented on examples/projections/misc/misc_eckertIV.py:
```diff
@@ -1,6 +1,6 @@
 r"""
-Eckert IV
-=========
+Eckert IV equal-area projection
+====================
```
```suggestion
Eckert IV equal-area projection
===============================
```

seisman reviewed (approved):
Looks good to me, except for two minor issues. Thanks for your patience on this PR.
